osf/management/commands/force_archive.py

```diff
@@ -280,48 +281,72 @@ def get_logs_to_revert(reg):
             Q(node=reg.registered_from) |
             (Q(params__source__nid=reg.registered_from._id) | Q(params__destination__nid=reg.registered_from._id))).order_by('-date')
 
+def get_file_obj_from_log(log, reg):
+    try:
+        return BaseFileNode.objects.get(_id=log.params['urls']['view'].split('/')[4])
+    except KeyError:
+        if log.action == 'osf_storage_folder_created':
+            return OsfStorageFolder.objects.get(
+                target_object_id=reg.registered_from.id,
```

Comment:
Minor: _Should_ additionally include `target_content_type_id=ContentType.objects.get_for_model(AbstractNode).id`, here and below. 

It's a `GenericRelation` without guaranteed uniqueness across `target_object_id` and `name`,  so this could potentially raise `MultipleObjectsReturned`. This is incredibly unlikely for `Folder` objects with the current datamodel, as only `AbstractNode` (and subclasses) and `Preprint` can have `.files`, and the only folder on Preprints should be the blank-named root folder. However, if the data model changes in the future, this could lead to increased risk of collision.